### PR TITLE
Enable navigation for code flow / thread flow locations

### PR DESCRIPTION
**Pull request by yongyan-gh**
# Descriptions
Result's codeflow / threadflow locations can have source file location data. User should be able to navigate to the corresponding location by double clicking the item in "Analysis Step" tab (the same behavior to navigate to source file by double clicking error list item). Currently this functionality is not added.

# Fixes
`AnalysisStepNode` (ThreadFlowLocation) is already derived from `CodeLocationObject` base class, resolving file path code already there for `AnalysisStepNode`. Just need to hook up with WPF event trigger.

I don't have a good way to unit test it. Will add Apex UI automation tests for it. Please let me know if any good idea about unit testing it.

Related to #458

### CODE REVIEW

File: src/Sarif.Viewer.VisualStudio.Core/Views/AnalysisSteps.xaml [hunk 1]
```diff
@@ -95,6 +95,10 @@
                                             <ColumnDefinition SharedSizeGroup="MessageGroup" />
                                             <ColumnDefinition SharedSizeGroup="LocationGroup" />
                                         </Grid.ColumnDefinitions>
+                                        <Grid.InputBindings>
+                                            <MouseBinding Gesture="LeftDoubleClick"
+                                                                  Command="{Binding NavigateCommand}"/>
```
Comment (eddynaka) [inline src/Sarif.Viewer.VisualStudio.Core/Views/AnalysisSteps.xaml:100]: >         [](http://example.com/codeflow?start=58&length=8)

can u align based on the others?

**Review by EasyRhinoMSFT — APPROVED** [review]
:shipit:

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: src/Sarif.Viewer.VisualStudio.Core/Views/AnalysisSteps.xaml (hunk 1 above)
Comment (yongyan-gh) [reply src/Sarif.Viewer.VisualStudio.Core/Views/AnalysisSteps.xaml]: fixed

**Comment by yongyan-gh** [comment]
opened issue for UI tests https://github.com/microsoft/sarif-internal/issues/18

**Pull request merged**